src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -1,16 +1,295 @@
 export const description = `
-Tests for GPUAdapter.requestDevice.
+Test GPUAdapter.requestDevice.
 
-TODO:
-- descriptor is {null, undefined}
-- descriptor with the combinations of features
+Note tests explicitly destroy created devices so that tests don't have to wait for GC to clean up
+potentially limited native resources.
 `;
 
+import { Fixture } from '../../../../common/framework/fixture.js';
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
-import { GPUTest } from '../../../gpu_test.js';
+import { getGPU } from '../../../../common/util/navigator_gpu.js';
+import { assert, raceWithRejectOnTimeout } from '../../../../common/util/util.js';
+import { kFeatureNames, kLimitInfo, kLimits } from '../../../capability_info.js';
+import { clamp, isPowerOfTwo } from '../../../util/math.js';
 
-export const g = makeTestGroup(GPUTest);
+export const g = makeTestGroup(Fixture);
 
-g.test('request_default_after_error')
-  .desc(`Request default device after a failed request with unsupported features.`)
-  .unimplemented();
+g.test('default')
+  .desc(
+    `
+    Test requesting the default device.
+    - No features listed in default device
+    - Default limits`
+  )
+  .fn(async t => {
+    const gpu = getGPU();
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+    const device = await adapter.requestDevice();
```

Comment:
nit: Since this is the same as the next test other than this line, how about making it a parameterization instead? Then we could also test some extra cases too. Something like this?

```ts
  .paramsSubcasesOnly(u =>
    u.combine('args', [
      [],
      [undefined],
      [{}],
      [{ requiredFeatures: [], requiredLimits: {} }],
    ] as const)
  )
  .fn(async t => {
    const { args } = t.params;

    const gpu = getGPU();
    const adapter = await gpu.requestAdapter();
    assert(adapter !== null);
    const device = await adapter.requestDevice(...args);
```

Reply:
Done